store/src/main/java/com/book/store/controller/BookController.java

```diff
@@ -77,14 +77,15 @@ public ResponseEntity<Void> deleteBook(@PathVariable Long id) {
     }
 
     @GetMapping("/search")
-    @Operation(summary = "Search books based on criteria")
-    public ResponseEntity<List<BookDto>> searchBooks(
+    @Operation(summary = "Search books based on criteria with pagination and sorting")
+    public ResponseEntity<Page<BookDto>> searchBooks(
```

Comment:
Page is a database object that is used for extracting partial data. 
Collection is used to transfer data. 

Replace return type with Collection, e.g. list

```suggestion
    public ResponseEntity<List<BookDto>> searchBooks(
```